### PR TITLE
Upstreamize Hammer guide

### PR DESCRIPTION
**Pull request by Lennonka**
Offering the Hammer guide to upstream.

* [x] I am okay with my commits getting squashed when you merge this PR.
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.12

### CODE REVIEW

File: guides/doc-Hammer_CLI/topics/con_introduction-to-hammer.adoc
Comment (Lennonka) [inline guides/doc-Hammer_CLI/topics/con_introduction-to-hammer.adoc:197]: I need help here. :)

File: guides/doc-Hammer_CLI/topics/con_introduction-to-hammer.adoc
Comment (ekohl) [inline guides/doc-Hammer_CLI/topics/con_introduction-to-hammer.adoc:122]: Might be good to add that the installer won't update the file and if you changed your credentials then you need to do that manually. You already have the instructions.

File: guides/doc-Hammer_CLI/topics/con_introduction-to-hammer.adoc
Comment (ekohl) [inline guides/doc-Hammer_CLI/topics/con_introduction-to-hammer.adoc:202]: I'd argue this instruction is wrong. It installs a single plugin to Hammer that ends up pulling in some others. For satellite I think the instruction should be to install `satellite-cli`, but I'm not 100% sure we publish that in our utils repo.

File: guides/doc-Hammer_CLI/topics/con_introduction-to-hammer.adoc
Comment (ekohl) [inline guides/doc-Hammer_CLI/topics/con_introduction-to-hammer.adoc:205]: I first thought the IP is wrong, but then think we can even reduce it further.
```suggestion
. Set the `:host:` entry in the `/etc/hammer/cli.modules.d/foreman.yml` file to the {Project} URL.
```

File: guides/doc-Hammer_CLI/topics/con_introduction-to-hammer.adoc
Comment (ekohl) [inline guides/doc-Hammer_CLI/topics/con_introduction-to-hammer.adoc:292]: This is specifically for [hammer-cli-foreman-admin](https://github.com/theforeman/hammer-cli-foreman-admin) which is a very unusual plugin. Normally Hammer is a tool that talks to the Foreman API over the network. hammer-cli-foreman-admin actually operates on the server itself. I'd love to get rid of it.

We already have a section that can change the log levels (https://docs.theforeman.org/nightly/Administering_Project/index-katello.html#Logging_and_Reporting_Problems_admin). I'd be my preference to use that instead as the single way to configure logging.

File: guides/doc-Hammer_CLI/topics/con_introduction-to-hammer.adoc
Comment (ekohl) [reply guides/doc-Hammer_CLI/topics/con_introduction-to-hammer.adoc:197]: For upstream it's the exact same repository as the server so you should be able to reuse that module

File: guides/doc-Hammer_CLI/topics/con_introduction-to-hammer.adoc
Comment (Lennonka) [reply guides/doc-Hammer_CLI/topics/con_introduction-to-hammer.adoc:202]: I checked on Ohsnap and yes, we publish the `satellite-cli` package in the Utils repo.

File: guides/common/attributes-titles.adoc
Comment (Lennonka) [inline guides/common/attributes-titles.adoc:10]: @apinnick I would like to have your opinion.

I want to change the title, but I'm not sure whether to call it _Hammer CLI reference_ or _Using Hammer CLI_. In downstream, we put it into the Reference category, but it contains procedures as well. What do you think?

**Comment by Lennonka** [comment]
I've made an attempt to rewrite the standalone installation procedure for upstream.
And incorporated all suggestions.

File: guides/common/attributes-titles.adoc
Comment (apinnick) [reply guides/common/attributes-titles.adoc:10]: For d/s, I would prefer "Hammer CLI guide" so that it is consistent with the existing "API Guide" (not sure why that title capitalizes "guide" -- will check). 

I would not use the contents  alone for determining the title. All API and CLI guides contain procedures for installing, configuring, setting log levels, etc. 

API and tool guides tend to be named either "Using the 'x' tool" or "X reference guide". Personally, I do not think it matters which format is used as long as the guide names are consistent. You could change the API guide to "Using the Satellite REST API" and call this guide "Using the Hammer CLI tool". :-)  

Consistency is the most important consideration, in my opinion.

Comment (Lennonka) [reply guides/common/attributes-titles.adoc:10]: I intend to change the title of the API guide too #3212 . I don't like having "guide" in the title, when we're avoiding it everywhere else.

Comment (apinnick) [reply guides/common/attributes-titles.adoc:10]: +1 I also don't like "guide" in the title.

**Comment by Lennonka** [comment]
I've moved the assembly and modules where they belong. I suggest merging it as is and making further changes in other PRs.

**Review by maximiliankolb — COMMENTED** [review]
* I noticed that many modules start with a level 2 heading. I am OK with this for now and can change it later in a follow-up PR.
* Unsure what "aqua-background" is for/ why "org" and "loc" has to be blue/yellow.
* Unsure if we _want_ to keep some rather long examples of stdout.

I had a glance at most changes but not all. I appreciate the effort to bring it upstream. All my suggestion can be applied by you or me after merging.

Merging now and improving later is perfectly fine for me. Any other concerns or approval?

**Comment by Lennonka** [comment]
> * Unsure what "aqua-background" is for/ why "org" and "loc" has to be blue/yellow.

It's sort of a flag that is introduced and explained in General information. It can be done diffetently, but for now I'm keeping it as is.

File: guides/common/assembly_hammer-reference.adoc
Comment (Lennonka) [inline guides/common/assembly_hammer-reference.adoc:1]: I'm not sure what to do with this file.

The problem is that the reference should be generated separately for Foreman, Katello, orcharhino and Satellite.
If I generate it in downstream and keep this file in upstream, our sync will overwrite it from upstream again. Therefore, it would be better to remove it in upstream.
However, if I don't include it in upstream, the build will not pass because it's required by `master.adoc`.

So the solution might be to exclude this file from syncing in downstream. Will you be able to do that as well, @maximiliankolb?

Comment (maximiliankolb) [reply guides/common/assembly_hammer-reference.adoc:1]: Yeah, I can handle this. No worries.

**Comment by maximiliankolb** [comment]
@Lennonka Just to ensure that I understand the intended workflow:

* for upstream Foreman and Foreman+Katello, we can check in the output `ref_*` files from `hammer full-help --md > hammer_full_help.md && ./generate-hammer-reference.sh hammer_full_help.md`
* for downstream, we can do the same on Satellite/orcharhino and check in those files to downstream-only branches?

I ran this on orcharhino and got 95 files in total. Do you envision generating the references files on a minimal installation, with select plugins, or with all available plugins in Satellite/orcharhino?

**Comment by Lennonka** [comment]
> * for upstream Foreman and Foreman+Katello, we can check in the output `ref_*` files from `hammer full-help --md > hammer_full_help.md && ./generate-hammer-reference.sh hammer_full_help.md`
> * for downstream, we can do the same on Satellite/orcharhino and check in those files to downstream-only branches?

That is correct.

> I ran this on orcharhino and got 95 files in total. Do you envision generating the references files on a minimal installation, with select plugins, or with all available plugins in Satellite/orcharhino?

In Satellite, we have all plugins enabled AFAIK, so that's a given. For Foreman and Foreman+Katello I would go for generating the full help on an instance with all Hammer plugins enabled, either with or without Katello plugin accordingly. I think it's better to include all plugins regardless of what the user has enabled. Better to have it and not use it than not have it when needed :)

File: guides/common/modules/con_introduction-to-hammer.adoc
Comment (maximiliankolb) [inline guides/common/modules/con_introduction-to-hammer.adoc:68]:
```suggestion
When running commands manually, {Team} recommends using the hammer authentication session and providing credentials with each command.
```

**Review by maximiliankolb — CHANGES_REQUESTED** [review]
One small request.

I had a look at the built docs without reading/testing everything. I think we can merge this as is and then improve later.

I will have a closer look at this as soon as I integrate this into downstream docs.

As there are no other comments, I say LGTM; please merge after a final tiny edit & green GHA.

Thanks again @Lennonka for this effort! :rocket:

**Comment by Lennonka** [comment]
Cherry-picked:
- cee50ae19d..c750bf7da3  3.12 -> 3.12

**Pull request merged**
